webapp/src/app.tsx

```diff
@@ -1864,15 +1864,16 @@ export class ProjectView
     }
 
     toggleTrace(intervalSpeed?: number) {
-        if (this.state.tracing) {
+        let tracing = this.state.tracing;
+        if (tracing) {
             this.editor.clearHighlightedStatements();
             simulator.setTraceInterval(0);
         }
         else {
             simulator.setTraceInterval(intervalSpeed != undefined ? intervalSpeed : simulator.SLOW_TRACE_INTERVAL);
```

Comment:
``!== undefined``

Reply:
ACtualy you could just do ``intervalSpeed || simulator.SLOW_TRACE_INTERNAL``

Reply:
Yep